src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -604,6 +604,30 @@ internal static bool IsRunningOnProcessorArchitectureARM()
 #endif
         }
 
+        /// <summary>
+        /// Get a temporary directory to use, needs to be unique to avoid collision
+        /// </summary>
+        internal static string GetTemporaryDirectory()
+        {
+            string tempDir = String.Empty;
+            string tempPath = Path.GetTempPath();
+            do
+            {
+                tempDir = Path.Combine(tempPath,System.Guid.NewGuid().ToString());
```

Comment:
Created doc bug https://github.com/PowerShell/PowerShell-Docs/issues/1126

I don't think users should be required to create those env vars, we should just work in most cases